src/com/facebook/buck/cli/TestCommand.java

```diff
@@ -605,6 +609,40 @@ public int compare(TestRule o1, TestRule o2) {
     return builder.build();
   }
 
+  @VisibleForTesting
+  Iterable<TestRule> filterUntestableRules(
+      Iterable<TestRule> testRules) {
+
+    ImmutableSortedSet.Builder<TestRule> builder =
+        ImmutableSortedSet.orderedBy(
+            new Comparator<TestRule>() {
+              @Override
+              public int compare(TestRule o1, TestRule o2) {
+                return o1.getBuildTarget().getFullyQualifiedName().compareTo(
+                    o2.getBuildTarget().getFullyQualifiedName());
+              }
+            });
+
+    for (TestRule rule : testRules) {
+      // xctool doesn't run XCUITests yet, but project generation works
+      boolean isAppleUITest = false;
+      if (rule instanceof AppleTest) {
+        AppleTest appleTestRule = (AppleTest)rule;
+        isAppleUITest = appleTestRule.isUiTest();
+        if (isAppleUITest) {
+          LOG.warn("Not running rule '%s' as XCUITests aren't supported.",
+              rule.getFullyQualifiedName());
+        }
+      }
+
+      if (!isAppleUITest) {
+        builder.add(rule);
+      }
+    }
+
+    return builder.build();
+  }
```

Comment:
I couldn't find a good way to not instantiate the `TestRule`, instead I propose to return an empty list of test steps for `ImmutableList<Step> runTests(...)`, with a dummy tests response that informs the user these tests have not been run.

The output of this is as follows:
```
RESULTS FOR //mock:TestApp
NOTESTS <100ms  0 Passed   0 Skipped   0 Failed   XCUITest runs not supported with buck test
```